File: TimeTableBackend/TimeTableBackend/Models/MonHoc.cs
using System;
using System.Collections;
using System.Collections.Generic;

namespace TimeTableBackend.Models
{
    public class MonHoc
    {
        public MonHoc()
        {
        }
        public int Id { get; set; }
        public string Ten { get; set; }
        public string MaMonHoc { get; set; }
        public int SoTinChi { get; set; }


        public virtual List<NhomMonHoc>? NhomMonHoc { get; set; }

        public int NienKhoaId { get; set; }
        public virtual NienKhoa NienKhoa { get; set; }
    }
}

File: TimeTableFrontEnd/src/app/shared/models/MonHoc.ts
export class Buoi {
  id: number;
  soTiet: number;
  batDauLuc: number;
  tietBatDau: number;
  giangVien: string;
  phong: string;
}

export class NhomMonHoc {
  id: string;
  phong: string;
  nmh: string;
  monHocId: number;
  monHoc: MonHoc;
  buois: Buoi[];
}

export class MonHoc{
  id: number;
  ten: string;
  maMonHoc: string;
  soTinChi: number;
  nhomMonHoc: NhomMonHoc[];
  isActive = false;
}
